qupath-core/src/test/java/qupath/lib/objects/classes/TestPathClassFactory.java

```diff
@@ -0,0 +1,143 @@
+ * %%
+ * QuPath is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * QuPath is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License 
+ * along with QuPath.  If not, see <https://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package qupath.lib.objects.classes;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.awt.Color;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import qupath.lib.common.ColorTools;
+
+@SuppressWarnings("javadoc")
+public class TestPathClassFactory {
+	
+	@Test
+	public void test_getPathClass1() {
+		sameClass(new PathClass("Ignore*", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.IGNORE));
+		sameClass(new PathClass("Image", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.IMAGE_ROOT));
+		sameClass(new PathClass("Immune cells", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.IMMUNE_CELLS));
+		sameClass(new PathClass("Necrosis", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.NECROSIS));
+		sameClass(new PathClass("Negative", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.NEGATIVE));
+		sameClass(new PathClass("Other", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.OTHER));
+		sameClass(new PathClass("Positive", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.POSITIVE));
+		sameClass(new PathClass("Region*", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.REGION));
+		sameClass(new PathClass("Stroma", ColorTools.makeRGB(180, 180, 180)), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.STROMA));
+		
+		// Can't use PathClass' constructor for 'Tumor' as getDerivedPathClass() creates an instance of 'Tumor' already.
+		sameClass(PathClassFactory.getPathClass("Tumor"), PathClassFactory.getPathClass(PathClassFactory.StandardPathClasses.TUMOR));
+	}
+	
+	@Test
+	public void test_getPathClass2() {
+		Assertions.assertThrows(IllegalArgumentException.class, () -> PathClassFactory.getPathClass("Class with" + System.lineSeparator() + "new line", Color.RED.getRGB()));
```

Comment:
Conceivably using `System.lineSeparator()` could give a test that works on Windows and fails on Linux, or vice versa. Consider instead adding two tests, with `\n` and `\r\n`